include/ddc/kernels/splines/spline_builder_2d_batched.hpp

```diff
@@ -0,0 +1,287 @@
+                    derivs_min2
+            = std::nullopt,
+            std::optional<
+                    ddc::ChunkSpan<double const, derivs_domain_type2, Layout, memory_space>> const
+                    derivs_max2
+            = std::nullopt,
+            std::optional<
+                    ddc::ChunkSpan<double const, derivs_domain_type, Layout, memory_space>> const
+                    mixed_derivs_min1_min2
+            = std::nullopt,
+            std::optional<
+                    ddc::ChunkSpan<double const, derivs_domain_type, Layout, memory_space>> const
+                    mixed_derivs_max1_min2
+            = std::nullopt,
+            std::optional<
+                    ddc::ChunkSpan<double const, derivs_domain_type, Layout, memory_space>> const
+                    mixed_derivs_min1_max2
+            = std::nullopt,
+            std::optional<
+                    ddc::ChunkSpan<double const, derivs_domain_type, Layout, memory_space>> const
+                    mixed_derivs_max1_max2
+            = std::nullopt) const;
+};
+
+
+template <
+        class ExecSpace,
+        class MemorySpace,
+        class BSpline1,
+        class BSpline2,
+        class IDimI1,
+        class IDimI2,
+        ddc::BoundCond BcXmin1,
+        ddc::BoundCond BcXmax1,
+        ddc::BoundCond BcXmin2,
+        ddc::BoundCond BcXmax2,
+        class... IDimX>
+template <class Layout>
+void SplineBuilder2DBatched<
+        ExecSpace,
+        MemorySpace,
+        BSpline1,
+        BSpline2,
+        IDimI1,
+        IDimI2,
+        BcXmin1,
+        BcXmax1,
+        BcXmin2,
+        BcXmax2,
+        IDimX...>::
+operator()(
+        ddc::ChunkSpan<double, spline_domain_type, Layout, memory_space> spline,
+        ddc::ChunkSpan<double const, vals_domain_type, Layout, memory_space> vals,
+        std::optional<ddc::ChunkSpan<double const, derivs_domain_type1, Layout, memory_space>> const
+                derivs_min1,
+        std::optional<ddc::ChunkSpan<double const, derivs_domain_type1, Layout, memory_space>> const
+                derivs_max1,
+        std::optional<ddc::ChunkSpan<double const, derivs_domain_type2, Layout, memory_space>> const
+                derivs_min2,
+        std::optional<ddc::ChunkSpan<double const, derivs_domain_type2, Layout, memory_space>> const
+                derivs_max2,
+        std::optional<ddc::ChunkSpan<double const, derivs_domain_type, Layout, memory_space>> const
+                mixed_derivs_min1_min2,
+        std::optional<ddc::ChunkSpan<double const, derivs_domain_type, Layout, memory_space>> const
+                mixed_derivs_max1_min2,
+        std::optional<ddc::ChunkSpan<double const, derivs_domain_type, Layout, memory_space>> const
+                mixed_derivs_min1_max2,
+        std::optional<ddc::ChunkSpan<double const, derivs_domain_type, Layout, memory_space>> const
+                mixed_derivs_max1_max2) const
```

Comment:
In the end I am not sure it is a good idea to make an out-of-line definition, this part is quite long

Reply:
What you mean by out-of-line ?

Reply:
I mean outside of the class, it is quite verbose to do that, especially for this function

Reply:
Ah yes, maybe we can change it later